src/coreComponents/mesh/mpiCommunications/CommunicationTools.cpp

```diff
@@ -1030,7 +1030,7 @@ bool CommunicationTools::asyncUnpack( MeshLevel & mesh,
   MpiWrapper::testSome( icomm.size(),
                         icomm.mpiRecvBufferRequest(),
                         &recvCount,
-                        &neighborIndices[0],
+                        &*neighborIndices.begin(),
```

Comment:
std::vector<>::begin() is undefined behavior if the vector is 0 length.

```suggestion
                        neighborIndices.data(),
```

Reply:
That works better. Once we move to C++20, we can ideally use things like `std::to_address(neighborIndices.begin())`.